vllm/config.py

```diff
@@ -969,16 +970,27 @@ def _maybe_override_draft_max_model_len(
 
     @staticmethod
     def create_draft_parallel_config(
-            target_parallel_config: ParallelConfig) -> ParallelConfig:
+            target_parallel_config: ParallelConfig,
+            speculative_tensor_parallel_size: Optional[int]) -> ParallelConfig:
         """Create a parallel config for use by the draft worker.
 
-        This is mostly a copy of the target parallel config. In the future the
-        draft worker can have a different parallel strategy, e.g. TP=1.
+        This is mostly a copy of the target parallel config, except the tp_size.
         """
+
+        speculative_tensor_parallel_size = (
+            speculative_tensor_parallel_size
```

Comment:
nit: if the user provides `--speculative-tensor-parallel-size 0`, this branch causes unexpected behavior. Can we explicitly guard against this?

Reply:
Thanks for the catch!
To prevent spec_tp to be set as target_tp when given spec_tp is 0, I've changed the code as below:
```
        if speculative_tensor_parallel_size is None:
            speculative_tensor_parallel_size = target_parallel_config.tensor_parallel_size
```

In addition, to prevent tp value from being 0, I think we need to make a separate PR to handle that case by adding a check in ParallelConfig._verify_args(). Because It seems to be the same in the `--tensor-parallel-size 0` case.

Reply:
i'll look at your new changes; no need to completely fix this (just a nit)